jena-arq/src/main/java/org/apache/jena/riot/RDFParserBuilder.java

```diff
@@ -83,7 +82,6 @@ public class RDFParserBuilder {
     // HTTP
     private String appAcceptHeader = null;
     private Map<String, String> httpHeaders = new HashMap<>();
-    private HttpClient httpClient = null;
```

Comment:
What's going on here?

It is important to be able to have custom HTTP setup because some SPARQL endpoints need specific settings.

Reply:
`httpClient` was only private field in `RDFParserBuilder` with no setter or way to modify or intercept it. 
It was always only initialized by:
`HttpClient clientJDK = ( httpClient != null ) ? httpClient : HttpEnv.getDftHttpClient();`
Since `HttpEnv` only initializes a static variable `HttpEnv.httpClient` once, storing a copy in the RDFParserBuilder is not necessary to prevent duplicated initialization.

In `RDFParserBuilder#build` it was passed to 'RDFParser' where there was only one private 'field' with no setter, no way to modfy, access or intercept it.
The only usage was in `RDFParserBuilder#openTypedInputStream(String urlStr, Path path)`, so I simply moved the initialization with `HttpEnv.getDftHttpClient()` there. 
That way, the initialization, wich took about 84ms, is only be performed, when a httpClient is needed. For reading files, this is typically not the case.

Reply:
I'd like to undertand what the cost actually is. You said "initialization" but I'm not sure what that refers to. if it parser specific or general? Which initialization step?

The Jena 5.1 error is that setter lines in `RDFParserBuilder` are commented out. 

Setting a custom `HttpClient` is important for external authentication mechanisms and complex setups. Jena does not cover all possible cases; it can't- it may involved extrat dependencies. Jena wraps up common patterns but leaving the ability to drop down to the base level.

(`HttpClient` is linked in the javadoc for `httpHeader` setting)

Reply:
![image](https://github.com/user-attachments/assets/1a773941-e165-47b8-9778-beb5daa6b397)
In this example you see, that for reading a small xml file from disk, which takes about 420ms the initialization of the httpClient, which is not used in that case, takes 108ms of that time.

> The Jena 5.1 error is that setter lines in RDFParserBuilder are commented out.

So, the setter `RDFParserBuilder#httpClient()`, which is currently commented out in Jena 5.2.0-SNAPSHOT should be made available again? 
Idea: We could pass the probably not initialized `httpBuilder` on to the `RDFParser` and deal there with the case that it might be null by reusing the initialization code which was used in the `RDFParserBuilder` before: `( httpClient != null ) ? httpClient : HttpEnv.getDftHttpClient();`. What do you think?

> (HttpClient is linked in the javadoc for httpHeader setting)

Which javadoc exactly do you mean? I could not find it.

Reply:
> > (HttpClient is linked in the javadoc for httpHeader setting)
> 
> Which javadoc exactly do you mean? I could not find it.

https://github.com/afs/jena/blob/452c8d287a930bb15474f3c30071aa572b27e87d/jena-arq/src/main/java/org/apache/jena/riot/RDFParserBuilder.java#L273

Reply:
> > > (HttpClient is linked in the javadoc for httpHeader setting)
> > 
> > 
> > Which javadoc exactly do you mean? I could not find it.
> 
> https://github.com/afs/jena/blob/452c8d287a930bb15474f3c30071aa572b27e87d/jena-arq/src/main/java/org/apache/jena/riot/RDFParserBuilder.java#L273

I see now. This setter is currrently commented out in the main branch for Jena 5.2.0-SNAPSHOT and the javadoc differs. 
I will apply the above mentioned solution in this PR.